### DISCUSSION TITLE
Dynamic Axis Scaling in Animated Charts

### DISCUSSION THREAD

**Discussion by philippulus** [d3/d3#3596] [Q&A] [answered]
Hi community,

I am looking for creating animated charts with dynamically scaling axis, changing X and Y values and moving symbols like in [this example](https://www.linkedin.com/posts/jameseagle_gold-finance-economics-activity-6987774486183845888-gtoy). 

https://user-images.githubusercontent.com/89039695/198820023-0b89600d-be44-4a6c-b882-31e1ed63b7a8.mp4

Looking at this [bar chart race example](https://observablehq.com/@d3/bar-chart-race) I get the impression it was made with d3js. Is this correct?

**Comment by curran** [comment, marked as answer]
Here's a snippet that does dynamic ticks based on a specified density (ticks per pixel):

```js
axisLeft(yScale).ticks(yTickDensity * (height - margin.bottom - margin.top))
```

Here's an old piece on bl.ocks.org that does this: https://bl.ocks.org/curran/6f97aa94357cae4611c54a80c11f6128
